pkg/restore/restore.go

```diff
@@ -694,6 +694,18 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 				continue
 			}
 
+			// Check if the PV exists in the cluster before attempting to create
+			// a volume from the snapshot, in order to avoid orphaned volumes (GH #609)
+			pv, err := resourceClient.Get(name, metav1.GetOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				ctx.log.Errorf("Error checking for existence of PV %s: %s", name, err)
+				continue
```

Comment:
Wondering if we should attempt the restore if we get an error here, rather than `continue`ing - WDYT?

Reply:
Ah, good point - we really should continue with the restore. I'll update.

Reply:
Please hold off for a minute

Reply:
If there's some other error besides not-found, it could be
- failure to reach the apiserver for some reason
- GET permission denied
- ???

The only one of these I'm concerned about, and hopefully this is extremely unlikely, is the ark serviceaccount is denied `GET persistentvolumes`. In this case, we probably shouldn't continue. WDYT?

Reply:
The permission denied would make sense. I assume that could be either of [`IsUnauthorized`](https://github.com/kubernetes/apimachinery/blob/release-1.11/pkg/api/errors/errors.go#L474) or [`IsForbidden`](https://github.com/kubernetes/apimachinery/blob/release-1.11/pkg/api/errors/errors.go#L480)? Is preferred over the other by the kube API server?

Reply:
I think I'm leaning more toward leaving the code as-is - if we can't successfully determine a PV's existence or non-existence, skip it.

Reply:
I'd like to have any errors show up the same way as for other resources (i.e. in the restore results).  I think as long as that's the case, either way is defensible.